src/SectorView.cpp

```diff
@@ -585,48 +586,60 @@ std::vector<SystemPath> SectorView::GetRoute()
-	RefCountedPtr<const Sector> target_sec = m_galaxy->GetSector(target);
+	const RefCountedPtr<const Sector> start_sec = m_galaxy->GetSector(start);
+	const RefCountedPtr<const Sector> target_sec = m_galaxy->GetSector(target);
 
 	// Get the player's hyperdrive from Lua, later used to calculate the duration between systems
-	ScopedTable hyperdrive = ScopedTable(LuaObject<Player>::CallMethod<LuaRef>(Pi::player, "GetEquip", "engine", 1));
+	const ScopedTable hyperdrive = ScopedTable(LuaObject<Player>::CallMethod<LuaRef>(Pi::player, "GetEquip", "engine", 1));
 	// Cache max range so it doesn't get recalculated every time we call GetDuration
-	float max_range = hyperdrive.CallMethod<float>("GetMaximumRange", Pi::player);
+	const float max_range = hyperdrive.CallMethod<float>("GetMaximumRange", Pi::player);
 
-	float dist = Sector::DistanceBetween(start_sec, start.systemIndex, target_sec, target.systemIndex);
+	const float dist = Sector::DistanceBetween(start_sec, start.systemIndex, target_sec, target.systemIndex);
 
-	int sec_dist = ceilf(dist / Sector::SIZE);
-
-	std::vector<SystemPath> nodes;
-	std::set<std::vector<SystemPath>::size_type> unvisited;
-
-	std::vector<float> path_dist; // distance from source to node
-	std::vector<std::vector<SystemPath>::size_type> path_prev; // previous node in optimal path
+	const int sec_dist = ceilf(dist / Sector::SIZE);
 
 	// nodes[0] is always start
+	std::vector<SystemPath> nodes;
 	nodes.push_back(start);
 
+	const Sint32 minX = std::min(start.sectorX, target.sectorX)-2, maxX = std::max(start.sectorX, target.sectorX)+2;
+	const Sint32 minY = std::min(start.sectorY, target.sectorY)-2, maxY = std::max(start.sectorY, target.sectorY)+2;
+	const Sint32 minZ = std::min(start.sectorZ, target.sectorZ)-2, maxZ = std::max(start.sectorZ, target.sectorZ)+2;
+	const vector3f start_pos = start_sec->m_systems[start.systemIndex].GetFullPosition();
+	const vector3f target_pos = target_sec->m_systems[target.systemIndex].GetFullPosition();
+
 	// go sector by sector for sec_dist sectors and add systems
 	// if they are within 110% of dist of both start and target
-	for (Sint32 sx = -sec_dist; sx <= sec_dist; sx++) {
-		for (Sint32 sy = -sec_dist; sy <= sec_dist; sy++) {
-			for (Sint32 sz = -sec_dist; sz < sec_dist; sz++) {
-				SystemPath sec_path = SystemPath(start.sectorX + sx, start.sectorY + sy, start.sectorZ + sz);
+	for (Sint32 sx = minX; sx <= maxX; sx++) {
+		for (Sint32 sy = minY; sy <= maxY; sy++) {
+			for (Sint32 sz = minZ; sz < maxZ; sz++) {
+				const SystemPath sec_path = SystemPath(sx, sy, sz);
 				RefCountedPtr<const Sector> sec = m_galaxy->GetSector(sec_path);
 				for (std::vector<Sector::System>::size_type s = 0; s < sec->m_systems.size(); s++) {
-					if (start.IsSameSystem(sec->m_systems[s].GetPath())) continue; // start is already nodes[0]
+					if (start.IsSameSystem(sec->m_systems[s].GetPath()))
+						continue; // start is already nodes[0]
+
+					bool isWithinLineSegment = false;
```

Comment:
Not used.

Reply:
Ah, it's actually used as an output parameter of a function.. Then it's ok to keep it then, or consider removing that altogether from DistanceFromLine() function